### PR TITLE
feat: support nuxt@3.0.0

### PR DESCRIPTION
**Pull request by NexZhu**
### Description

This PR make `unplugin-vue-components` compaible with the new `nuxt@3.0.0` release, this is needed until Nuxt supports using resolvers from this plugin, in order to use e.g. `ElementPlusResolver` in Nuxt projects.

### Linked Issues

https://github.com/nuxt/framework/issues/487

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,11 +76,15 @@
   },
   "peerDependencies": {
     "@babel/parser": "^7.15.8",
+    "@nuxt/kit": "^3.0.0",
```
Comment (sxzz) [inline package.json:79]: Not really necessary I think.

File: package.json [hunk 2]
```diff
@@ -99,6 +103,8 @@
     "@antfu/eslint-config": "^0.33.1",
     "@babel/parser": "^7.20.5",
     "@babel/types": "^7.20.5",
+    "@nuxt/kit": "^3.0.0",
```
Comment (sxzz) [inline package.json:106]: We should put it to dependencies instead of dev deps

**Review by sxzz — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (NexZhu) [reply package.json]: done

File: src/nuxt.ts [hunk 3]
```diff
@@ -1,16 +1,15 @@
+import { addVitePlugin, addWebpackPlugin, defineNuxtModule } from '@nuxt/kit'
+// Workaround for:
+// src/nuxt.ts(5,1): error TS2742: The inferred type of 'default' cannot be named without a reference to '.pnpm/@nuxt+schema@3.0.0_rollup@2.79.0/node_modules/@nuxt/schema'. This is likely not portable. A type annotation is necessary.
+import type { NuxtModule } from '@nuxt/schema'
```
Comment (sxzz) [inline src/nuxt.ts:4]:
```suggestion
import type {} from '@nuxt/schema'
```

File: src/nuxt.ts [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
 import type { Options } from './types'
 import unplugin from '.'
 
-export default function (this: any, options: Options) {
-  // install webpack plugin
-  this.extendBuild((config: any) => {
-    config.plugins = config.plugins || []
-    config.plugins.unshift(unplugin.webpack(options))
-  })
+const _: NuxtModule | undefined = undefined
```
Comment (sxzz) [inline src/nuxt.ts:8]:
```suggestion
```

File: src/nuxt.ts (hunk 3 above)
Comment (NexZhu) [reply src/nuxt.ts:4]: Maybe leave the comment for future reference?

Comment (sxzz) [reply src/nuxt.ts:4]: It's ok

**Comment by NexZhu** [comment]
@sxzz Updated

**Review by sxzz — APPROVED** [review]

**Pull request merged**